### PR TITLE
PSA unified driver specification: minor updates

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
These are minor updates following @ronald-cron-arm's feedback in #3493.

These are mostly editorial changes. The semantic changes are:
* Rename the `"functions"` JSON property to `"entry_points"`.
* Rename the `derive_key` entry point to `key_derivation_output_key`.
* The `"entry_points"` property is now mandatory in capabilities.
* Transparent drivers are now considered in the order in which they are passed in the driver description list, rather than in an unspecified order.
* Algorithm specifications may use algorithm wildcard policies.

### CODE REVIEW

File: docs/proposed/psa-driver-interface.md
Comment (ronald-cron-arm) [inline docs/proposed/psa-driver-interface.md:91]: Typo: "capabilit"

File: docs/proposed/psa-driver-interface.md
Comment (danh-arm) [inline docs/proposed/psa-driver-interface.md:637]: I guess in theory a platform could have multiple accelerators with an intersecting set of mechanisms. The drivers for those accelerators may come from different sources and it could be a pain for the system integrator if this were considered an error. In the fallback case, you say "The core considers drivers in the order given by the driver description list" so I think we should do the same here too.

Comment (gilles-peskine-arm) [reply docs/proposed/psa-driver-interface.md:639]: Updated. I think this may not be fully settled, so I'm keeping the entry in the “open questions” section for now.

File: docs/proposed/psa-driver-interface.md
Comment (danh-arm) [inline docs/proposed/psa-driver-interface.md:641]: > ...but may mask integration errors.

The implementation could emit a warning to mitigate this.

File: docs/proposed/psa-driver-interface.md
Comment (danh-arm) [inline docs/proposed/psa-driver-interface.md:635]: minor: Given this seems to be a tentative plan, it might be worth keeping it in the "Open Questions" section. The main part of the spec should use strong language.

**Review by danh-arm — APPROVED** [review]
Will you squash these commits before final merge to make the history a bit prettier?

File: docs/proposed/psa-driver-interface.md
Comment (gilles-peskine-arm) [reply docs/proposed/psa-driver-interface.md:635]: This is in the open questions section.

Comment (danh-arm) [reply docs/proposed/psa-driver-interface.md:635]: Sorry, I got a bit lost where I was looking

**Comment by gilles-peskine-arm** [comment]
I don't intend to squash commits: why would I? It would make it harder to understand the history, it would disrupt the link between the review and the history, and it would have no benefit. History is supposed to be useful, not pretty.

**Comment by gilles-peskine-arm** [comment]
This has one approval. It's only documentation, and that documentation is marked as a work in progress, so I'm going ahead and merging. Feedback on the whole document is still welcome, just open an issue or discuss on the [mailing list](https://lists.trustedfirmware.org/mailman/listinfo/psa-crypto).

**Pull request merged**
